internal/pkg/warewulfd/parser.go

```diff
@@ -44,8 +47,17 @@ func parseReq(req *http.Request) (parserInfo, error) {
 		ret.efifile = path_parts[2]
 	}
 	ret.hwaddr = hwaddr
+    ipaddrtemp = ""
+    ipaddr = strings.Split(req.RemoteAddr, ":")
+    // IPv6 addresses hextets are split on ":"
+    if len(ipaddr) > 2 {
+       for i := 0; i < len(ipaddr)-1; i++ {
+            ipaddrtemp += ipaddr[i] + ":"
+       }
+       ret.ipaddr6 = strings.TrimSuffix(ipaddrtemp, ":")
+    }
 	ret.ipaddr = strings.Split(req.RemoteAddr, ":")[0]
```

Comment:
Should `ret.ipaddr` not be set if `len(ipaddr) > 2`? What effect would this have elsewhere? What effect would it have for `ret.ipaddr` to be set to the first part of an ipv6 address (which seems to be what happens otherwise)?

Maybe something like

```
parts := string.Split(req.RemoteAddr, ":")
if len(parts) > 2 {
    ret.ipaddr6 = strings.Join(parts[:len(parts)-1], ":")
} else {
    ret.ipaddr = parts[0]
}
```